### ISSUE TITLE
Improve Vaadin project detection

### ISSUE THREAD

**Issue by MarcinVaadin** [vaadin/intellij-plugin#33]
Currently there is simple `String::contains` search of `com.vaadin` inside `pom.xml` / `build.gradle` to detect Vaadin project. It might cause lot of false positives. For parsing pom.xml `maven-model` might be used. For gradle file there is `gradle-tooling-api` but it's bit more complex to use.

Also there is no multi module detection of Vaadin child projects.

**Comment by MarcinVaadin** [comment]
Update:

On multimodule projects, root module might not contain Vaadin dependency - Vaadin plugin is not started.

**Comment by Artur-** [comment]
Note that as part of the "Debug as hotswap", parsing of some build data for both Maven and Gradle were added so you can do e.g.
```
var deps: List<MavenArtifact> = MavenProjectsManager.getInstance(module.project).findProject(module)!!.dependencies
```

and also find a parent project
```
var parentId: MavenId = MavenProjectsManager.getInstance(module.project).findProject(module)!!.parentId!!;
MavenProjectsManager.getInstance(module.project).findProject(parentId)
```

**Comment by Artur-** [comment]
Was this fixed by #82 ?

**Comment by MarcinVaadin** [comment]
Yes
